### ISSUE TITLE
Upgrade 3.12 to 3.13

### ISSUE THREAD

**Issue by nickcoons** [lizardfs/lizardfs#922]
I have a cluster of several servers running 3.12 (they also provide other tasks, like a GaleraDB cluster, and a few others). My plan is to replace one server at a time. The old servers are running Debian 10, and the new ones are Debian 11. So that means that the LizardFS version will differ as well.

When I search for upgraded LizardFS, I can't find any documentation that specifically addresses these versions. So a few questions:

- Is there an upgrade path for 3.12 to 3.13?
- Are the two versions compatible? Meaning, can two different versions co-exist within the same cluster?

Regarding the second question, as I'm replacing one server at a time.. the first replaced server is a chunkserver and client, as is the third, and the second server scheduled for replacement is also the master. So, can I replace the first server (thus, having one server running 3.13 chunkserver and client) while the remaining servers run 3.12, and continuing that strategy to replace one server at a time? Or do I need to upgrade the various services in a specific order? I know for 2.6.0 to 3.9.4 there was a specific process, but that's a much more major jump, so I'm not sure if that still applies.

**Comment by Antuan-AE** [comment]
Hello nick, do you have any files with EC goals inside your cluster?

**Comment by nickcoons** [comment]
No, all of them are just normal copied chunks. Ah, yes.. I think I read about a bug where upgrading from 3.12 to 3.13 with EC caused data loss.. is that right?

**Comment by szycha76** [comment]
Right: #746 :-(

**Comment by nickcoons** [comment]
Since I don't have any EC goals, can I get some details on the upgrade process as mentioned in my first post?

**Comment by wildtv** [comment]
I'd like some clarification here as well.
The 3.12 builds/  packages.lizardfs.com seem to be broken now, and the GPG key is missing.
Debian seems to keep a 3.10 version on repo, the dev.lizardfs.com seems to carry 3.13 for bullseye.
I've got a 3.12 cluster and I'd like to bring in another machine as a chunkserver, but it's looking like my easiest install is 3.13. 
I don't want to create issues, so knowing  things will be backwards compatible 3.13 chunkserver -> 3.12 master would be handy.

**Comment by onlyjob** [comment]
> Debian seems to keep a 3.10 version on repo

The latest version in Debian "Buster" a.k.a. "oldstable" is **3.12**, which is the last release without upgrade path, because LizardFS has been removed from Debian in favour of MooseFS. Neither current "stable" (Bullseye) nor upcoming "testing" have LizardFS.

 * https://tracker.debian.org/pkg/lizardfs
 * https://packages.debian.org/search?keywords=lizardfs

I can't comment about the state of _vendor_ packages, that were always inferior to properly Debianised packages.

**Comment by wojciesh** [comment]
> I'd like some clarification here as well. The 3.12 builds/ packages.lizardfs.com seem to be broken now, and the GPG key is missing. Debian seems to keep a 3.10 version on repo, the dev.lizardfs.com seems to carry 3.13 for bullseye. I've got a 3.12 cluster and I'd like to bring in another machine as a chunkserver, but it's looking like my easiest install is 3.13. I don't want to create issues, so knowing things will be backwards compatible 3.13 chunkserver -> 3.12 master would be handy.

Hi **wildtv**, please contact me at _**vitos [a.t] lizardfs.com**_, we will be happy to have a meeting with you. We would like to better understand your case and provide you with the best solution.

**Comment by smurfix** [comment]
@wojciesh Seriously? The last material commit was 10 months ago.

At this point the best solution is to drop lizardfs like the hot potato it obviously is.
